internal/clients/currency/curency_beacon_api.go

```diff
@@ -0,0 +1,88 @@
+package currency_client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"subscription-api/config"
+	"subscription-api/internal/clients"
+	"subscription-api/pkg/utils"
+)
+
+const (
+	CurrencyBeaconAPIBasePath string = "https://api.currencybeacon.com/v1/"
+	CurrencyBeaconAPILabel    string = "Currency Beacon API https://currencybeacon.com/"
+)
+
+type currencyBeaconAPIClient struct {
+	httpClient *clients.HTTPClient
+	apiKey     string
+	log        responseLogger
+}
+
+func NewCurrencyBeaconAPIClient(httpClient *clients.HTTPClient, apiKey string, logger config.Logger) *currencyBeaconAPIClient {
+	return &currencyBeaconAPIClient{
+		httpClient: httpClient,
+		apiKey:     apiKey,
+		log:        buildResponseLogger(logger, CurrencyBeaconAPILabel),
+	}
+}
+
+// Gets latest exchange rates for specified currencies.
+func (c *currencyBeaconAPIClient) Convert(
+	ctx context.Context,
+	baseCcy string,
+	targetCcies []string,
+) (map[string]float64, error) {
+	resp, err := c.httpClient.Get(ctx, fmt.Sprintf("%slatest?api_key=%s&base=%s", CurrencyBeaconAPIBasePath, c.apiKey, baseCcy))
```

Comment:
I'm a little confused that I don't see the connection closing in all clients. Maybe you do it somewhere else? Can you clarify this for me?

Reply:
Connection to what should be closed?

Reply:
something like  resp.Body.Close()

Reply:
Updated code base by closong all of occurences of http.Response.Body